### ISSUE TITLE
gpac sefaults trying to parse an m4a file

### ISSUE THREAD

**Issue by finnhughes** [gpac/gpac#1863]
- [y] I looked for a similar issue and couldn't find any.
- [y] I tried with the latest version of GPAC. Installers available at http://gpac.io/downloads/gpac-nightly-builds/
- [y] I give enough information for contributors to reproduce my issue

My m4a causes gpac to segfaults, for my use case this is easily worked around by using an aac file instead (many thanks for adding udta support to gpac 🥇 and making this easy).
```
$ gpac -i TOS-enAD.m4a inspect
Segmentation fault: 11
```

The inspect filter is only an example it segfaults when trying to remux too, we can see that MP4Box has no such problem:
```
$ MP4Box -info TOS-enAD.m4a
# Movie Info - 1 track - TimeScale 1000
Duration 00:00:12.032
Fragmented: no
Progressive (moov before mdat)
Major Brand M4A  - version 1 - compatible brands: isom M4A  mp42
Created: GMT Thu Jul 15 14:59:03 2021

File has root IOD (9 bytes)
Scene PL 0xff - Graphics PL 0xff - OD PL 0xff
Visual PL: No visual capability required (0xff)
Audio PL: AAC Profile @ Level 2 (0x29)
No streams included in root OD

# Track 1 Info - ID 1 - TimeScale 48000
Media Duration 00:00:12.074
Track has 1 edits: track duration is 00:00:12.032
Media Info: Language "English (eng)" - Type "soun:mp4a" - 566 samples
2 UDTA types:
	tagc: public.accessibility.describes-video
	name: English (describes video)
Alternate Group ID 1
MPEG-4 Audio AAC LC (AOT=2 implicit) - 2 Channel(s) - SampleRate 48000
	RFC6381 Codec Parameters: mp4a.40.2
	All samples are sync
	Max sample duration: 1024 / 48000
```

The debug log doesn't look very helpful to me, it looks like it happily gets a long way then faults:
```
[Mutex Filtermp4dmxTasks] At 146 Released by thread Main Process
Segmentation fault: 11
```

I've attached the problematic m4a, the debug log and the aac file I used to workaround the problem

https://www.mediafire.com/file/a62ijsh2o78kle1/TOS-enAD.m4a/file
https://www.mediafire.com/file/q8k6la50sz425ok/debug.log/file
https://www.mediafire.com/file/xj6lb0pz5aetngw/TOS-enAD.aac/file

**Comment by jeanlf** [comment]
Did you try latest HEAD ? This was fixed by a37a0ae

**Comment by finnhughes** [comment]
Thanks, I was a commit behind
